python/cugraph/centrality/betweenness_centrality_wrapper.pyx

```diff
@@ -26,82 +27,227 @@ import cudf
+        raise ValueError("result_dtype can only be np.float64 or np.float32")
+
+    return result_df
+
+def run_mg_work(input_data, normalized, endpoints,
+                weights, sources,
+                result_dtype, session_id):
+    result = None
+    # 1. Get session information
+    session_state = worker_state(session_id)
+    number_of_workers = session_state["nworkers"]
+    worker_idx = session_state["wid"]
+
+    # 2. Get handle
+    handle = session_state['handle']
+
+    # 3. Get Batch
+    batch = get_batch(sources, number_of_workers, worker_idx)
+    total_number_of_sources = len(sources)
+
+    result = run_internal_work(handle, input_data, normalized,
+                               endpoints, weights, batch,
+                               total_number_of_sources, result_dtype)
+    return result
+
+
+def mg_batch_betweenness_centrality(input_graph, normalized, endpoints,
+                                    weights, vertices, result_dtype):
+    df = None
+    client = get_client()
+    comms = Comms.get_comms()
+    replicated_adjlists = input_graph.mg_batch_adjlists
+    work_futures =  [client.submit(run_mg_work,
+                                   (data, type(input_graph)
+                                   is DiGraph),
+                                   normalized,
+                                   endpoints,
+                                   weights,
+                                   vertices,
+                                   result_dtype,
+                                   comms.sessionId,
+                                   workers=[worker]) for
+                    idx, (worker, data) in enumerate(replicated_adjlists.items())]
+    dask.distributed.wait(work_futures)
+    df = work_futures[0].result()
+    return df
+
+
+def sg_betweenness_centrality(input_graph, normalized, endpoints, weights,
+                              vertices, result_dtype):
+    total_number_of_sources = len(vertices)
+    handle = cugraph.raft.common.handle.Handle()
+    adjlist = input_graph.adjlist
+    input_data = ((adjlist.offsets, adjlist.indices, adjlist.weights),
+                  type(input_graph) is cugraph.DiGraph)
+    df = run_internal_work(handle, input_data, normalized, endpoints, weights,
+                           vertices, total_number_of_sources, result_dtype)
+    return df
 
+
+# NOTE: The current implementation only has <int, int, float, float> and
+#       <int, int, double, double> as explicit template declaration
+#       The current BFS requires the GraphCSR to be declared
+#       as <int, int, float> or <int, int double> even if weights is null
+def betweenness_centrality(input_graph, normalized, endpoints, weights,
+                           vertices, result_dtype):
+    """
+    Call betweenness centrality
+    """
+    df = None
+    comms = Comms.get_comms()
+    if not input_graph.adjlist:
+        input_graph.view_adj_list()
+    if comms and input_graph.mg_batch_enabled == True:
```

Comment:
We can utilize https://github.com/rapidsai/cugraph/blob/38b032efa630ae01e29b8c73b81c34bfa3f5a905/python/cugraph/comms/comms.py#L44

Reply:
Updated in 09cbb8c.